File: src/main/java/com/apiApp/CateringFacilityAPI/model/jpa/FacilityLocation.java
package com.apiApp.CateringFacilityAPI.model.jpa;

import com.fasterxml.jackson.annotation.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "fac_location")
public class FacilityLocation extends BaseEntity{

    @NotNull
    private String name;

    @NotNull
    private String city;

    @NotNull
    private String address;

    @NotNull
    @ManyToOne
    @JsonIgnore
    private Facility facility;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Facility getFacility() {
        return facility;
    }

    @JsonProperty
    public void setFacility(Facility facility) {
        this.facility = facility;
    }
}

File: frontend/src/app/model/FacilityLocation.ts
import {Facility} from "./Facility";
import {BaseEntity} from "./BaseEntity";

export class FacilityLocation  extends BaseEntity{

  constructor(id?: number, name?: string, city?: string, address?: string, facility?: Facility);

  constructor(id: number, name: string, city: string, address: string, facility: Facility) {
    super(id);
    this.name = name;
    this.city = city;
    this.address = address;
    this.facility = facility;
  }

  public name: string;
  public city: string;
  public address: string;
  public facility: Facility;
}
